### ISSUE TITLE
Can "codemap.json" be described by dedicated mapper?

### ISSUE THREAD

**Issue by niciki-niciki** [oleg-shilo/codemap.vscode#102]
Hi,

Can we make our own dedicated mapper for JSON?

Thanks,
Daniel

**Comment by oleg-shilo** [comment]
Yes, you can. Just follow the [instructions](https://github.com/oleg-shilo/codemap.vscode/wiki/Adding-custom-mappers#mapping-with-a-dedicated-mapper) for the dedicated mapper.

YOu can overwrite any mapper that is distributed with the extension. So it should work right away fr you.

**Comment by niciki-niciki** [comment]
There is a warning for
"codemap.json": "/path/to/my/mapper_json.js",

Incorrect type. Expected "array".

**Comment by niciki-niciki** [comment]
I have just discovered that the same dedicated mapper "mapper_json.js" works as expected for .json_test files, but not for .json files (the same file but different extension):

"codemap.json": "/path/to/my/mapper_json.js", // this does not work for the same file content, but extension .json

"codemap.json_test": "/path/to/my/mapper_json.js", // this works OK for the same file content, but extension .json_test

**Comment by oleg-shilo** [comment]
I found it. 
The extension actually works. 
The very same problem was reported in https://github.com/oleg-shilo/codemap.vscode/issues/44 and it was fixed. The expected functionality is achieved via setting the `overloaded` suffix:

```json
"codemap.overloaded.md": "D:\\Dropbox_local\\software\\V\\Visual Studio Code\\extensions\\mapper_md.js",
```

The problem is... I did not describe the solution for #44 in the documentation. It was my mistake. I apologize for that.

I even remember that I implemented something for that but I could not find any mention in the documentation. And only when I started debugging your test case I saw the fix.

I have updated the documentation now: https://github.com/oleg-shilo/codemap.vscode/wiki/Adding-custom-mappers#important
